soc/arm64/nxp_imx/mimx9/mmu_regions.c

```diff
@@ -8,6 +8,30 @@
 #include <zephyr/devicetree.h>
 #include <zephyr/sys/util.h>
 
+/**
+ * @brief Auto generate mmu region entry for node_id
+ *
+ * Example usage:
+ *
+ * @code{.c}
+ *      DT_FOREACH_STATUS_OKAY_VARGS(nxp_imx_gpio,
+ *				  MIMX9_DT_MMU_REGION_FLAT_ENTRY,
+ *				 (MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_NS))
+ * @endcode
+ *
+ * @note  Since devicetree_generated.h does not include
+ *        node_id##_P_reg_FOREACH_PROP_ELEM* definitions,
+ *        we can't automate dts node with multiple reg
+ *        entries.
+ *
+ */
+#define MIMX9_DT_MMU_REGION_FLAT_ENTRY(node_id, attrs)  \
+	MMU_REGION_FLAT_ENTRY(DT_NODE_FULL_NAME(node_id), \
+			      DT_REG_ADDR(node_id), \
+			      DT_REG_SIZE(node_id), \
+			      attrs),
```

Comment:
This macro is used to generage an entry in `mmu_regions[]` down below. Take this for example:

```c
DT_FOREACH_STATUS_OKAY_VARGS(nxp_kinetis_lpuart,
			  MIMX9_DT_MMU_REGION_FLAT_ENTRY,
			 (MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_NS))
```

It will generate mmap entries for all LPUART instances with `status = "okay"` set automatically.

In the past we need to create LPUART1 ... LPUART8 manually, and if one of them are disabled in DT, the mmap entry is not removed. This results in a huge array of unused regions and occupy space in mmu translate table.

The `DT_FOREACH_STATUS_OKAY_VARGS` macro only takes token withthis format `(node_id, VA_ARGS)` , so make the `DT_FOREACH` macro happy.

Our most (if not all) drivers has not been updated to use `runtime_device_mmio` APIs, thus these regions are needed to map them before drivers actually access the region. This is common to all our MPU products (i.MX8M, i.MX93, etc.) I will assume these are introduced before `runtime_device_mmio` APIs.

If we update drivers to map their own registers during initliztion, (LPI2C driver as an example in this pr), then this is no longer needed.

Reply:
I understand your point. But the transition to `device_mmio` APIs takes time. Adding this macro does not create a new way of mapping regions, but just enhances the `mmu_regions.c` so that existing drivers can be easily enabled on MPU, without need to hack the driver or adding a pile of (maybe unused) regions into that array.